larreco/ClusterFinder/ClusterAna_module.cc

```diff
@@ -115,611 +138,471 @@ namespace cluster {
       if (fSkipCosmics && theTruth->Origin() == simb::kCosmicRay) continue;
-      if (theTruth->Origin() == simb::kBeamNeutrino) isNeutrino = true;
-      if (theTruth->Origin() == simb::kSingleParticle) isSingleParticle = true;
-      float KE = 1000 * (part->E() - part->Mass());
-
-      // Get the neutrino track ID. Assume that there is only one neutrino
-      // interaction and it is first in the list of BeamNeutrino particles
-      if ((isNeutrino || isSingleParticle) && neutTrackID < 0) {
-        neutTrackID = trackID;
-        if (isNeutrino) {
-          simb::MCNeutrino theNeutrino = theTruth->GetNeutrino();
-          neutEnergy = 1000. * theNeutrino.Nu().E();
-          neutIntType = theNeutrino.InteractionType();
-          neutCCNC = theNeutrino.CCNC();
-        }
-
-        for (unsigned short iv = 0; iv < recoVtxList.size(); ++iv) {
-          recoVtxList[iv]->XYZ(xyz);
-          fNuVtx_dx->Fill(part->Vx() - xyz[0]);
-          fNuVtx_dy->Fill(part->Vy() - xyz[1]);
-          fNuVtx_dz->Fill(part->Vz() - xyz[2]);
-        } // iv
-      }   // theTruth->Origin() == simb::kBeamNeutrino && neutTrackID <
-
-      if (fPrintLevel > 3)
-        mf::LogVerbatim("ClusterAna") << "Origin " << theTruth->Origin() << " trackID " << trackID
-                                      << " PDG " << part->PdgCode() << " KE " << (int)KE << " MeV "
-                                      << " neutTrackID " << neutTrackID << " Mother "
-                                      << part->Mother() << " Proc " << part->Process();
-
-      bool isCharged = (pdg == 11) || (pdg == 13) || (pdg == 211) || (pdg == 321) || (pdg == 2212);
-
-      if (!isCharged) continue;
-
-      // KE (MeV) cuts
-      if (pdg == 11) {
-        if (fElecKERange[0] < 0) continue;
-        // only allow primary electrons
-        if (part->Process() != "primary") continue;
-        if (KE < fElecKERange[0] || KE > fElecKERange[1]) continue;
-      }
-      if (pdg == 13) {
-        if (fMuonKERange[0] < 0) continue;
-        if (KE < fMuonKERange[0] || KE > fMuonKERange[1]) continue;
+      float TMeV = 1000 * (mcp.E() - mcp.Mass());
+      // consider electrons, muons, pions, kaons and protons
+      int pdgIndx = PDGCodeIndex(mcp.PdgCode());
+      if (pdgIndx < 0) continue;
+      bool useIt = true;
+      if (pdgIndx == 0) {
+        if (fElecKERange[0] < 0 || TMeV < fElecKERange[0] || TMeV > fElecKERange[1]) useIt = false;
+      } else if (pdgIndx == 1) {
+        if (fMuonKERange[0] < 0 || TMeV < fMuonKERange[0] || TMeV > fMuonKERange[1]) useIt = false;
+      } else if (pdgIndx == 2) {
+        if (fPionKERange[0] < 0 || TMeV < fPionKERange[0] || TMeV > fPionKERange[1]) useIt = false;
+      } else if (pdgIndx == 3) {
+        if (fKaonKERange[0] < 0 || TMeV < fKaonKERange[0] || TMeV > fKaonKERange[1]) useIt = false;
+      } else if (pdgIndx == 4) {
+        if (fProtKERange[0] < 0 || TMeV < fProtKERange[0] || TMeV > fProtKERange[1]) useIt = false;
```

Comment:
After using the `ParticleInfo` structure(s) suggested elsewhere, this can become:
```.cpp
auto const& info = info_for_particle[pdgIndx];
if (info.fKERange[0] < 0 || TMeV < info.fKERange[0] || TMeV > info.fKERange[1]) useIt = false;
```